packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -377,10 +383,20 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
 
   void _updateScaffolds() {
     for (final ScaffoldState scaffold in _scaffolds) {
-      scaffold._updateSnackBar();
+      if (_isRoot(scaffold))
+        scaffold._updateSnackBar();
     }
   }
 
+  // Nested Scaffolds are handled by the ScaffoldMessenger by only presenting a
+  // SnackBar in the root Scaffold of the nested set.
+  bool _isRoot(ScaffoldState scaffold) {
+    final ScaffoldState? parent = scaffold.context.findAncestorStateOfType<ScaffoldState>();
+    if (parent != null && _scaffolds.contains(parent))
```

Comment:
We cannot cross the scope of the current ScaffoldMessenger, otherwise, there will be problems when there are multiple scopes.

Reply:
Sorry, I am mistaken, only the scaffold of the current scope is in the collection. :)